Tabs: restore vertical alignment for tabs content

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

This PR re-introduces some layout styles to `Tabs.Tab` so that the tab contents are vertically centered

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Fixing an involuntary regression

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

By re-adding flexbox styles to the Tab container

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

- Open the "With Tab Icons and Tooltips" Storybook example
- Verify the vertical alignment of the icons, make sure the icons are vertically centered

Note: the centering fix can't be seen when the tabs only contain text, as that's already working as expected.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

| Before (trunk) | After (this PR) |
|---|---|
| ![Screenshot 2024-10-17 at 18 08 53](https://github.com/user-attachments/assets/2c59a297-9a02-45cb-b91d-efe6577465e0) | ![Screenshot 2024-10-17 at 18 09 14](https://github.com/user-attachments/assets/b279db9b-855b-4b8b-b6ee-40069afeb4a1) |

ciampo committed:
Tabs: restore vertical alignent for tabs content

ciampo committed:
CHANGELOG

tyxla commented on packages/components/src/tabs/styles.ts:
```diff
@@ -234,6 +234,9 @@ export const Tab = styled( Ariakit.Tab )`
 	flex-grow: 1;
+
+	display: flex;
+	align-items: center;
```
It seems that tabs with icons that have custom wider widths will have icons that aren't centered:

![Screenshot 2024-10-18 at 11 45 42](https://github.com/user-attachments/assets/9f847a3b-bcc5-4be1-93a7-23c02ebc781c)

Do we also need a `justify-content: center`:


```suggestion
	align-items: center;
	justify-content: center;
```

tyxla reviewed (changes requested):
This seems to introduce a regression with tabs that are wider than the icons.

ciampo committed:
Fix horizontal alignment

ciampo committed:
Remove unnecessary style overrides

ciampo commented:
@tyxla feedback addressed:
- fixed horizontal alignment for both horizontal and vertical orientation: https://github.com/WordPress/gutenberg/pull/66215/commits/c780c0da98ed6c1672b8d1fbbb69be7d4945028d
- removed some unnecessary style overrides for the specific example that you [shared above](https://github.com/WordPress/gutenberg/pull/66215#discussion_r1806126888): https://github.com/WordPress/gutenberg/pull/66215/commits/42142442f1836fe9524035eab646f280b1b2acd6

tyxla commented on packages/block-editor/src/components/block-inspector/style.scss:
```diff
@@ -46,9 +46,3 @@
 	border-top: $border-width solid $gray-300;
 }
-
-.block-editor-block-inspector__tab-item {
```
Thanks for the cleanup 👍

tyxla commented on packages/block-editor/src/components/inspector-controls-tabs/utils.js:
```diff
@@ -9,21 +9,18 @@ export const TAB_SETTINGS = {
 	title: __( 'Settings' ),
 	value: 'settings',
 	icon: cog,
-	className: 'block-editor-block-inspector__tab-item',
```
Good riddance 👍 

https://wpdirectory.net/search/01JAFMC50Q107D92KTX11G52FY

tyxla reviewed (approved):
Looks good, tests well 👍 🚀 

Thanks @ciampo 🚢
